src/lib/containers/SingleNewsFeedPost.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { useState } from 'react' // importing FunctionComponent
+import RssFeed from './RssFeed'
+
+export type SingleNewsFeedPostProps = {
+  feedUrl: string 
+}
+
+export default function SingleNewsFeedPost(props: SingleNewsFeedPostProps) {
+  const [modifiedFeedUrl, setModifiedFeedUrl] = useState()
+  const {feedUrl} = props
+  let updatedModifiedFeedUrl = feedUrl
+  const urlParams = new URLSearchParams(window.location.search)
+  if (urlParams) {
+    urlParams.forEach((value, key) => {
+      updatedModifiedFeedUrl = `${updatedModifiedFeedUrl}&${key}=${value}`
+    })
+  }
+  if (updatedModifiedFeedUrl && modifiedFeedUrl !== updatedModifiedFeedUrl)
```

Comment:
I think state can be avoided all together here, should be able to use updatedModifiedFeedURL below and remove the modifiedFeedURL variable